core/src/lib.rs

```diff
@@ -136,7 +129,7 @@ where
         &self,
         user_data: F::UserDataPtr,
         operation: packet::Operation,
-    ) -> Result<Packet<'_, F::UserDataPtr>, AcquirePacketError> {
+    ) -> Packet<'_, F::UserDataPtr> {
```

Comment:
@50U10FCA7 I think it makes sense to use the same approach as @jeffutter used in #19. We had this `.acquire()` method to reflect the necessity of calling `sys::tb_client_acquire_packet()` and so reflecting the undergoing logic of what's going on. However, since packets are now fully client-allocatable and there is no such thing as their acquiring, why still mimicking this with `.acquire()` method then? Let's just use `Packet::new()` method instead.